geps/gep-1619.md

```diff
@@ -163,43 +163,299 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
+- Name=_value_
 - Expires=_date_
 - Max-Age=_number_
 - Domain=_domain_
 - Path=_path-value_
 - Secure
 - HttpOnly
 
-However, there are de-facto standard attributes that have been proposed as updates to [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2):
-- SameSite=_[Strict|Lax|None]_ ([Draft RFC Update](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-cookie-same-site-00))
-- Partitioned ([Draft RFC Update](https://www.ietf.org/archive/id/draft-cutler-httpbis-partitioned-cookies-01.html))
-
-**Session Cookie Strategy**
-
-Session cookie strategy is configuration specific to the creation of the cookie by the dataplane. These are not specific configuration parameters of the cookie itself, but instead the operation to take when inserting the cookie such as:
-- Rewriting an existing cookie value
-- Inserting a new cookie
-- Prefixing an existing cookie
-- Preserving an existing cookie value
-- Only adding cookies on specific HTTP methods (POST/GET)
+Other cookie attributes not defined by RFC6265, but are captured in draft RFCs and could be considered de facto
+standards due to wide acceptance are:
+- SameSite=[Strict|Lax|None]
+- Partitioned
+
+We implement core support for the `Name`, `Path`, and `TTL` attributes as the metrics in [implementations](#Implementations)
+for session persistence shows these being the most widely adopted and supported. Subsequently, we proceed with `Domain`,
+`SameSite`, `HttpOnly`, and `Secure` as extended support, considering their partial acceptance among APIs. The
+`Partitioned` attribute lacks support in any implementation API and will be omitted. Similarly, since the `Expires`
+attribute is redundant compared to the preferred `Max-Age` attribute, it will also be skipped.
+
+```go
+// SessionPersistencePolicy provides a way to define session persistence rules
+// for a service.
+//
+// Support: Core
+type SessionPersistencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    
+    // Spec defines the desired state of SessionPersistencePolicy.
+    Spec SessionPersistencePolicySpec `json:"spec"`
+    
+    // Status defines the current state of SessionPersistencePolicy.
+    Status SessionPersistencePolicyStatus `json:"status,omitempty"`
+}
+
+// SessionPersistencePolicySpec defines the desired state of
+// SessionPersistencePolicy.
+// Note: there is no Override or Default policy configuration.
+type SessionPersistencePolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // HttpRoutes, GRPCRoute, Services are the only valid API target 
+    // references.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+    
+    // HttpCookie defines and configures a cookie to achieve
+    // session persistence. 
+    HttpCookie *HttpCookie `json:"httpCookie"`
+}
+
+// HttpCookie defines a cookie to achieve session persistence.
+//
+// Support: Core
+type HttpCookie struct {
```

Comment:
@costinm pls see my comment elsewhere. Applications using this feature need to know the name of the cookie used so they can copy that specific cookie from a response to the next requests. If the cookie name is not meant to be configurable in this API then how will applications know the name? Do implementations get to use the name they will use for the cookie and the applications then become non-portable across implementations?

Reply:
@costinm are suggesting to remove `HTTPCookie` and replace it with an alternative design, such as a enum at the `spec` level and leave `duration` at the `spec` level too? For example:
```
// SessionPersistenceSpec defines the desired state of
// SessionPersistence.
// Note: there is no Override or Default policy configuration.
type SessionPersistenceSpec struct {
    // TargetRef identifies an API object to apply policy to.
    // HttpRoutes, GRPCRoute, Services are the only valid API target 
    // references.
    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
    
    // Type defines  the method to achieve session persistence.
    Type *SessionPersistenceType `json:"type"`

    // Duration defines the session persistence duration 
    // 
    // Support: Core
    //
    // +optional
    Duration int64 `json:"ttl,omitempty"`
}

// SessionPersistenceType defines a session persistence type.
type SessionPersistenceType string

const (
    // SessionPersistenceTypeHttpCookie specifies to use a HTTP cookie for session persistence
    SessionPersistenceTypeHttpCookie SessionPersistenceType = "HTTPCookie"
)

```

And if I'm understanding, what happens if we add a SessionPersistence `Type` of `Headers` in the future? Wouldn't the `Duration` field be inappropriate for this other type of session persistence?

Reply:
> Wouldn't the Duration field be inappropriate for this other type of session persistence?

Good point. So having a separate `HTTPCookie` object is preferred and allows us to make changes in the future

Reply:
Application code is supposed to copy ALL cookies from the response - based on the normal rules.
It does not need to know the names. 

LBs and infrastructure may insert cookies for other purpose, backends like Tomcat or PHP may generate their own cookies - clients should not be restricted or forced to ONLY copy the cookie named in the Gateway API and not use any other form of persistent session.

Duration of a session has nothing to do with Cookies - you can have a 30 min duration for a session with 
a long lived cookie or headers or TLS session ID. 

In all cases I know duration is about the session idle, in rare cases about the total session time - but almost never about the low-level implementation details or how long the cookie will last. Usually cookies are refreshed when are close to expiration - we should make it VERY clear that cookie duration and session duration are not the same thing ( and session duration and idle timeout are also very different).

Reply:
A cookie of max-age 10 mins cannot be used for a session lasting more than 10 mins because the client side cookie jar will not keep it around more than 10 mins. Did I miss something?

Reply:
> Duration of a session has nothing to do with Cookies

Like Sanjay is asking, I think "nothing" here might be a strong word because if you are using cookies for session persistence and the cookie expires, your session will expire too. But this makes sense interpreted as "it's an implementation detail".

So essentially you are saying the `Duration` should set the session idle timeout. That makes sense, but so far, I've been designing the API based on implementation's / data planes existing supported configuration with their own APIs / configs. e.g. Envoy can set [name and TTL](https://www.envoyproxy.io/docs/envoy/latest/api-v3/type/http/v3/cookie.proto#envoy-v3-api-msg-type-http-v3-cookie) for a cookie. However, I haven't come across an implementation that has a generic session idle timeout configuration. Is Envoy able to support this?

My question boils down to: is it okay that  implementations will have to implement new functionality to support duration? Or am I misunderstanding the suggestion?